modules/tree/src/tree_application.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 Obolenskiy Arseniy
+#include <cstring>
+#include <limits>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include "include/tree_application.h"
+
+std::string TreeApplication::Help(const char *appname) {
+    return "This is application for tree class\n"
+           "Usage:\n"
+           "> " + std::string(appname) + " actions that described below\n"
+           "Actions:\n"
+           "add <value> - add value to the tree\n"
+           "del <value> - delete value from the tree\n"
+           "search <value> - find the value in the tree\n"
+           "clear - delete the whole tree\n";
+}
+
+std::string TreeApplication::operator()(int argc, const char** argv) {
+    if (argc == 1) {
+        return Help(*argv);
+    }
+    try {
+        for (int i = 1; i < argc; ) {
+            int offset = ParseOperation(argv + i);
+            i += offset;
+        }
+        return out_.str();
+    } catch (std::runtime_error &re) {
+        return re.what();
+    }
+}
+
+int TreeApplication::ParseNumber(const char *s) {
+    int len = strlen(s);
+    if ((s[0] != '-' && !isdigit(s[0])) || (s[0] == '-' && len == 1))
+        throw std::runtime_error(std::string(s) + " is not the number");
+    for (int i = 1; i < len; ++i)
+        if (!isdigit(s[i]))
+            throw std::runtime_error(std::string(s) + " is not the number");
+    int64_t result = strtol(s, nullptr, 10);
```

Comment:
```suggestion
    int64_t result = std::strtol(s, nullptr, 10);
```
и #include <cstdlib>